src/terminal-window.c

```diff
@@ -3852,16 +3928,23 @@ static void
 view_menubar_toggled_callback (GtkToggleAction *action,
                                TerminalWindow *window)
 {
+    G_GNUC_BEGIN_IGNORE_DEPRECATIONS;
     terminal_window_set_menubar_visible (window, gtk_toggle_action_get_active (action));
+    G_GNUC_END_IGNORE_DEPRECATIONS;
 }
 
 static void
 view_fullscreen_toggled_callback (GtkToggleAction *action,
                                   TerminalWindow *window)
 {
+    int toggle_action_state;
```

Comment:
https://developer.gnome.org/gtk3/stable/GtkToggleAction.html#gtk-toggle-action-get-active

I think it must be `gboolean` instead `int`

Documentation says:

> Returns the checked state of the toggle action.

I think is better other name, something like `toggle_action_checked` for example

Reply:
I tried gboolean first, but it gives me a build warning which say that int is the alias for it.
Maybe you have another result?

Reply:
Btw. i found a post at stackflow where someone says that gboolean is a integer in C.

Reply:
sure, i can change it to `toggle_action_checked` when first concern is resolved.

Reply:
I didn't get any new warnings after changing `int` to `gboolean`...

Reply:
> I tried gboolean first, but it gives me a build warning which say that `int` is the alias for it.

That's a very odd sentence, if `gboolean` is an alias of `int` (which it *is*), there is no reason why a compiler would complain about it.  Are you sure the warning was really about `gboolean` itself, and not about something else and it was trying to help you giving you the underlying type of the value for debugging purposes?

And regardless of the fact that `int` and `gboolean` are actually the same type I agree that `gboolean` should be used here, because it's semantically more correct.

Reply:
I see no warnings with these changes:
```patch
diff --git a/src/terminal-window.c b/src/terminal-window.c
index 82e0f21..389b969 100644
--- a/src/terminal-window.c
+++ b/src/terminal-window.c
@@ -3937,14 +3937,14 @@ static void
 view_fullscreen_toggled_callback (GtkToggleAction *action,
                                   TerminalWindow *window)
 {
-    int toggle_action_state;
+    gboolean toggle_action_checked;
 
     g_return_if_fail (gtk_widget_get_realized (GTK_WIDGET (window)));
 
     G_GNUC_BEGIN_IGNORE_DEPRECATIONS;
-    toggle_action_state = gtk_toggle_action_get_active (action);
+    toggle_action_checked = gtk_toggle_action_get_active (action);
     G_GNUC_END_IGNORE_DEPRECATIONS;
-    if (toggle_action_state)
+    if (toggle_action_checked)
         gtk_window_fullscreen (GTK_WINDOW (window));
     else
         gtk_window_unfullscreen (GTK_WINDOW (window));